src/v2/controllers/vreplicaset_controller/trusted/liveness_theorem.rs

```diff
@@ -33,18 +33,63 @@ pub open spec fn owned_selector_match_is(vrs: VReplicaSetView, obj: DynamicObjec
     &&& obj.metadata.deletion_timestamp.is_None()
 }
 
-// TODO: the current not_interfered_by invariant is radically strong. Weaken it later.
 pub open spec fn vrs_not_interfered_by(other_id: int) -> StatePred<ClusterState> {
     |s: ClusterState| {
         forall |msg| {
             &&& #[trigger] s.in_flight().contains(msg)
             &&& msg.content.is_APIRequest()
             &&& msg.src == HostId::Controller(other_id)
         } ==> match msg.content.get_APIRequest_0() {
-            APIRequest::CreateRequest(req) => req.obj.kind != Kind::PodKind,
-            APIRequest::UpdateRequest(req) => req.obj.kind != Kind::PodKind,
-            APIRequest::UpdateStatusRequest(req) => req.obj.kind != Kind::PodKind,
-            APIRequest::DeleteRequest(req) => req.key.kind != Kind::PodKind,
+            // Other controllers don't create pods owned by a VReplicaSet.
+            APIRequest::CreateRequest(req) => !{
+                let owner_references = req.obj.metadata.owner_references.get_Some_0();
+                &&& req.obj.kind == Kind::PodKind
+                &&& req.obj.metadata.namespace.is_Some()
+                &&& req.obj.metadata.owner_references.is_Some()
+                &&& exists |i: int| 0 <= i < owner_references.len() ==>
+                        #[trigger] owner_references[i].kind == VReplicaSetView::kind()
```

Comment:
@codyjrivera We also need to say that VRS is the **controller** owner reference which is unique. Consider using this `pod.metadata().owner_references_contains(v_replica_set.controller_owner_ref())` here: https://github.com/anvil-verifier/anvil/blob/main/src/v2/controllers/vreplicaset_controller/exec/reconciler.rs#L410

Reply:
We don't have access to a singular `vrs` object in the non-interference property. We want the other controllers not to interfere with _any_ replicaset.

Perhaps
```
exists |vrs : VReplicaSetView| #[trigger] owner_references.contains(vrs.controller_owner_ref())
```
can express what we want here? The two statements are obviously not equivalent (look at the definition of controller_owner_ref, https://github.com/anvil-verifier/anvil/blob/45eb50563566d38b3347b82d0a1b37f410025791/src/v2/controllers/vreplicaset_controller/trusted/spec_types.rs#L23, to see that the former statement on lines 49-50 does not imply the latter definition I just gave).

Was that the gap you wanted me to plug?

Reply:
> exists |vrs : VReplicaSetView| #[trigger] owner_references.contains(vrs.controller_owner_ref())

yes this should work